Added Prehook option to prepare method

edmundw314 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#41863 Added Prehook option to prepare method**

Differential Revision: [D22671288](https://our.internmc.facebook.com/intern/diff/D22671288)

edmundw314 committed:
Added Prehook option to prepare method

[ghstack-poisoned]

edmundw314 committed:
Update on "Added Prehook option to prepare method"


Differential Revision: [D22671288](https://our.internmc.facebook.com/intern/diff/D22671288)

[ghstack-poisoned]

edmundw314 committed:
Update on "Added Prehook option to prepare method"


Differential Revision: [D22671288](https://our.internmc.facebook.com/intern/diff/D22671288)

[ghstack-poisoned]

edmundw314 committed:
Update on "Added Prehook option to prepare method"


Differential Revision: [D22671288](https://our.internmc.facebook.com/intern/diff/D22671288)

[ghstack-poisoned]

edmundw314 committed:
Update on "Added Prehook option to prepare method"


Differential Revision: [D22671288](https://our.internmc.facebook.com/intern/diff/D22671288)

[ghstack-poisoned]

edmundw314 committed:
Update on "Added Prehook option to prepare method"


Differential Revision: [D22671288](https://our.internmc.facebook.com/intern/diff/D22671288)

[ghstack-poisoned]

edmundw314 committed:
Update on "Added Prehook option to prepare method"


Differential Revision: [D22671288](https://our.internmc.facebook.com/intern/diff/D22671288)

[ghstack-poisoned]

edmundw314 committed:
Update on "Added Prehook option to prepare method"


Differential Revision: [D22671288](https://our.internmc.facebook.com/intern/diff/D22671288)

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/quantize.py:
```diff
@@ -73,7 +73,11 @@ def _observer_forward_hook(self, input, output):
 
-def add_observer_(module, non_leaf_module_list=None, device=None):
+def _observer_forward_pre_hook(self, input):
+    self.activation_pre_process(input[0])
```
why is this `input[0]` can you add a comment? also maybe you can remove `return None` and add a comment why we don't need to return as well

jerryzh168 reviewed (approved):
Looks good, please add comments

edmundw314 committed:
Update on "Added Prehook option to prepare method"


Differential Revision: [D22671288](https://our.internmc.facebook.com/intern/diff/D22671288)

[ghstack-poisoned]

edmundw314 committed:
Update on "Added Prehook option to prepare method"


Differential Revision: [D22671288](https://our.internmc.facebook.com/intern/diff/D22671288)

[ghstack-poisoned]
